### DISCUSSION TITLE
native validation

### DISCUSSION THREAD

**Discussion by ranran2121** [react-hook-form/react-hook-form#7292] [Q&A] [answered]
Hi,
I am currently facing some issues with form validation:
I have a form with several inputs and 3 of them are mandatory(ie. two inputs of type text, and one input from a selection)
I am using this snippet
```
const { register, handleSubmit } = useForm({ shouldUseNativeValidation: true });
```
and then this other one on the required inputs
```
{...register("firstName", { required: "Please enter your first name." })} // custom message
````
however :
1-when I try to submit the form without the first input text, -> the input box gets highlighted and I am not able to submit but the custom message does NOT appear
2-when I try to submit the form without the second input text, -> the input box gets highlighted and I am not able to submit and the custom message does appear

can someone explain to me this behaviour and how can I make ALL the inputs mandatory before submitting, please?

**Comment by bluebill1049** [comment, marked as answer]
can you share a codesandbox?

**Reply by ranran2121** [reply]
I do not know how to do it :(
May I add some more of my code here?

**Reply by bluebill1049** [reply]
can you use this template: https://codesandbox.io/s/elegant-spence-y5q8t?

**Reply by ranran2121** [reply]
here it is
https://codesandbox.io/s/elegant-spence-y5q8t?file=/src/App.js

**Reply by bluebill1049** [reply]
did you save?

**Reply by ranran2121** [reply]
try now

**Reply by ranran2121** [reply]
@bluebill1049 have you seen it?

**Reply by bluebill1049** [reply]
<img width="1211" alt="Screen Shot 2021-12-30 at 8 31 40 pm" src="https://user-images.githubusercontent.com/10513364/147739310-c3e733ab-2e01-46d7-845a-c98cf86dd3aa.png">

https://codesandbox.io/s/elegant-spence-y5q8t?file=/src/App.js

pleases check yourself first...  (it's just a default hook form template)

**Reply by ranran2121** [reply]
forgive me, you asked me to make a copy of my code (which I did). I am asking why my code is not showing an error message whereas I have inserted the desired option in the correct form
````
{...register("surname", {
                required: "please, enter the surname"
              })}
````

what I get is that the form is not submitted and the input field is highlighted but no message appears. Could you please check my code and explain to me how to get the message? Thank you
https://codesandbox.io/s/awesome-sound-2hmln?file=/src/App.js:1374-1471

**Reply by bluebill1049** [reply]
can you reproduce the issue here: https://codesandbox.io/s/elegant-spence-y5q8t?file=/src/App.js:0-759 as it's working for me.

**Reply by ranran2121** [reply]
I cannot reproduce the issue on your code, because the same syntax in your code is working. But if you go on my code
https://codesandbox.io/s/awesome-sound-2hmln?file=/src/App.js

and try to submit the form without filling the required inputs (name, surname, email, department) you just have the input box borders highlighted but no message. Are you able to see my code in the link I provided, please?

**Reply by bluebill1049** [reply]
why are you doing this? 

`onClick={handleSubmit(onSubmits)}` that's the line of issue

**Reply by ranran2121** [reply]
ok, done it...thank you :)
